src/models/user.model.ts

```diff
@@ -0,0 +1,79 @@
+import mongoose, { Document, Model } from 'mongoose';
+import validator from 'validator';
+import bcrypt from 'bcryptjs';
+import { toJSON } from './plugins/toJSON.plugin';
+import { paginate } from './plugins/paginate.plugin';
+import { Role, roles } from '../config/roles';
+import { prop, getModelForClass, pre, plugin, DocumentType } from '@typegoose/typegoose';
```

Comment:
In my opinion, I find that "Typegoose" brings more complexity and less flexibility to the code. Unless I'm mistaken I don't have the impression that the types work well at the level of services (examples with User in services/user.services) but this is surely due to a bad manipulation on my part.